### DISCUSSION TITLE
FXP v5

### DISCUSSION THREAD

**Discussion by amitguptagwl** [NaturalIntelligence/fast-xml-parser#586] [General]
I have completed the development FXP v5 3 weeks ago. However, due to the other responsibilities, I'm not able to complete the tests and some remaining changes. The plan is to push a v5 branch soon if performance test are promising. So if possible I can get the community support to write the tests faster.

**Features**
 * Support for streams
 * Improve tag path expression for future features
 * Improvement in HTML Parsing
 * Support custom value parser as plugin
 * More user control on parsed output with less configuration
 * Will reveal more [watch this thread]

**Comment by amitguptagwl** [comment]
Due toe the common code base for stream and string, the performance of new XML parser has been degraded drastically that I was not expecting

Here is the comparison when parsing few KBs XML file.

fxp v4 - preserve order : 22344.761675388618 requests/second
fxp v5 - preserve order : 5713.2193482794 requests/second

Though, I'll work on improving the performance, what do you think what is more important?
* 👀 Support for streams
* 🚀 Parsing Speed

**Reply by jochenschmich-aeberle** [reply]
🚀 Parsing Speed for me anytime
May I kindly ask, if v5 is on its way to release?

**Reply by amitguptagwl** [reply]
v5 is available for experimental use with the current version. Please check the docs once. No plan for separate release currently.

**Reply by ArifRathod** [reply]
How i can test v5? is there any link where I can install it ?

**Reply by amitguptagwl** [reply]
https://github.com/NaturalIntelligence/fast-xml-parser/tree/master/docs/v5

**Comment by amitguptagwl** [comment]
New stats are satisfactory

v4_arr: 2.537ms
v4_obj: 0.866ms
v5_orderedArr: 2.006ms
v5_minarr: 0.779ms
v5_obj: 0.652ms

**Comment by amitguptagwl** [comment]
v5 is renamed to v6

**Comment by amitguptagwl** [comment]
Released as separate package

[@nodable/flexible-xml-parser](https://github.com/nodable/flexible-xml-parser)
